### PR TITLE
HTTP API Gateway with a hello world lambda integration

### PR DESCRIPTION
**Pull request by BarrettOlson**
Summary
---------

This creates an API gateway with terraform and sends request to a new hello world lambda.
Currently added IAM permission checks to the API so the only get data from the API is with an authenticated user. 

Test Plan
---------

Probably easiest shown through a demo... 
To test you'll need to create an IAM user if you don't have one already (be sure to notes its SecretKey) and then follow the steps [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-use-postman-to-call-api.html) to add the requested values to Postman.

### CODE REVIEW

File: hasher-matcher-actioner/hmalib/lambdas/api/status_api.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# DYNAMODB_TABLE = os.environ["DYNAMODB_TABLE"]
+
+
+def lambda_handler(event, context):
+    """
+    Status of delpoyed HMA 
+    """
+    # logger.info("Received event: " + json.dumps(event, indent=2))
+
+    return {
```
Comment (schatten) [inline hasher-matcher-actioner/hmalib/lambdas/api/status_api.py:25]: We should look for a library that makes these things easier.

File: hasher-matcher-actioner/terraform/api-surface/main.tf [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    {
+      Name = "StatusAPILambdaRole"
+    }
+  )
+}
+
+data "aws_iam_policy_document" "status_api" {
+  statement {
+    effect    = "Allow"
+    actions   = ["dynamodb:GetItem"]
```
Comment (schatten) [inline hasher-matcher-actioner/terraform/api-surface/main.tf:75]: Love this! Secure by default!

File: hasher-matcher-actioner/terraform/api-surface/main.tf [hunk 3]
```diff
@@ -0,0 +1,205 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
```
Comment (eriksendc) [inline hasher-matcher-actioner/terraform/api-surface/main.tf:1]: Yea... another naming comment! Why api-surface vs just api? For webapp I used webapp not webapp-surface or ui-surface. Seems like we need to pick one.

**Review by schatten — APPROVED** [review]
LGTM.

File: hasher-matcher-actioner/hmalib/lambdas/api/status_api.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import json
+
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# s3_client = boto3.client("s3")
+# dynamodb  = boto3.resource("dynamodb")
+
+# DYNAMODB_TABLE = os.environ["DYNAMODB_TABLE"]
```
[the remark this answers is not in the document]
Comment (BarrettOlson) [reply hasher-matcher-actioner/hmalib/lambdas/api/status_api.py:16]: later commit

File: hasher-matcher-actioner/terraform/api-surface/main.tf [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+  role          = aws_iam_role.status_api.arn
+  image_uri     = var.lambda_docker_info.uri
+  image_config {
+    command = [var.lambda_docker_info.commands.status_api]
+  }
+  timeout     = 300
+  memory_size = 512
+  environment {
+    variables = {
+      DYNAMODB_TABLE = var.datastore.name
```
[the remark this answers is not in the document]
Comment (BarrettOlson) [reply hasher-matcher-actioner/terraform/api-surface/main.tf:39]: that is the commented out part for a later commit

File: hasher-matcher-actioner/terraform/main.tf [hunk 6]
```diff
@@ -125,3 +125,25 @@ resource "aws_sqs_queue_policy" "pdq_hasher_queue" {
   queue_url = aws_sqs_queue.pdq_images_queue.id
   policy    = data.aws_iam_policy_document.pdq_hasher_queue.json
 }
+
+
+# Connect Hashing Data to API
+
+module "api_surface" {
```
Comment (eriksendc) [inline hasher-matcher-actioner/terraform/main.tf]: If we go with just `api` then probably change this, too (though you've probably already dealt with this...)

**Review by eriksendc — APPROVED** [review]

**Pull request merged**